src/ValidatedDTO.php

```diff
@@ -391,4 +401,25 @@ private function isOptionalProperty(string $property): bool
 
         return in_array('optional', $propertyRules) || in_array('nullable', $propertyRules);
     }
+
+    private function parseData(?array $data): array
+    {
+        if (is_null($data)) {
+            return [];
+        }
+
```

Comment:
Call the `mapBeforeValidation` here and assign to variable to avoid calling it multiple times